src/Task.h

```diff
@@ -704,7 +704,6 @@ class Task {
     std::string prname;
     std::vector<struct user_desc> thread_areas;
     remote_ptr<struct syscallbuf_hdr> syscallbuf_child;
-    std::vector<uint8_t> syscallbuf_hdr;
```

Comment:
There were a number of cases where syscallbuf_hdr was accessed after the child mapping was unmapped. I didn't get to look into whether any of those were necessary. If they are not, I think replacing syscallbuf_hdr is feasible. There is still a question of whether it is desirable, since we access it as a data structure in a number of places. It might make sense to have some template magic that allows transparent access to a data structure through a remote_ptr. Not sure. Will have to look into this more.